platform/darwin/src/MGLImageSource.h

```diff
@@ -0,0 +1,59 @@
+#import "MGLSource.h"
+
+#import "MGLFoundation.h"
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+@interface MGLImageSource : MGLSource
+
+#pragma mark Initializing a Source
+
+/**
+ Returns an image source with an identifier and coordinates
+
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @return An initialized image source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns an image source with an identifier, coordinates and a URL
+ 
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @param url An HTTP(S) URL, absolute file URL, or local file URL relative to the
+    current application’s resource bundle.
+ @return An initialized shape source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates URL:(NSURL *)url;
+
+/**
+ Returns an image source with an identifier, coordinates and an image
+ 
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @param image The image to display for the sourcde.
+ @return An initialized shape source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates image:(MGLImage *)image;
+
+#pragma mark Accessing a Source’s Content
+
+/**
+ The URL to the source image.
+
+ If the receiver was initialized using `-initWithIdentifier:coordinates:` or 
+ `-initWithIdentifier:coordinates:Image:`, this property is set to `nil`.
+ */
+@property (nonatomic, copy, nullable)NSURL *URL;
+
+@property (nonatomic, nullable, setter=setImage:)MGLImage *image;
```

Comment:
I think it’s fine for the property to have a getter with the caveat that it remains unset if the source is created from a URL. For precedent, [`MGLTileSource.configurationURL`](https://www.mapbox.com/ios-sdk/api/3.6.0-beta.2/Classes/MGLTileSource.html#/c:objc%28cs%29MGLTileSource%28py%29configurationURL) is unset if the raster or vector source was created using tile URL templates instead of a TileJSON URL.